Applications/Quickstarts.Servers/Alarms/AlarmHolders/AlarmHolder.cs

```diff
@@ -0,0 +1,485 @@
+            }
+
+            return hasBranches;
+        }
+
+        public BaseEventState GetBranch(byte[] eventId)
+        {
+            BaseEventState state = null;
+            string eventIdString = Utils.ToHexString(eventId);
+
+            ConditionState alarm = m_alarm as ConditionState;
+            if (alarm != null)
+            {
+                state = alarm.GetBranch(eventId);
+            }
+
+            return state;
+        }
+
+        public void ClearBranches()
+        {
+            ConditionState alarm = m_alarm as ConditionState;
+            if (alarm != null)
+            {
+                alarm.ClearBranches();
+                m_alarmController.SetBranchCount(0);
+            }
+        }
+
+        public void GetBranchesForConditionRefresh(List<IFilterTarget> events)
+        {
+            ConditionState alarm = m_alarm as ConditionState;
+            if (alarm != null)
+            {
+                Dictionary<string, ConditionState> branches = alarm.GetBranches();
+                foreach (BaseEventState branch in branches.Values)
+                {
+                    events.Add(branch);
+                }
+            }
+        }
+
+
+        protected virtual void CreateBranch()
+        {
+        }
+
+        private void InitializeInternal(BaseEventState alarm, NodeId branchId = null)
+        {
+            string triggerNodeId = (string)m_parent.NodeId.Identifier + "." + TriggerName;
+
+            string alarmName = AlarmName/* + GetBranchNodeIdString(branchId)*/;
+            string alarmNodeId = (string)m_parent.NodeId.Identifier + "." + AlarmName;
+
+            alarm.SymbolicName = alarmName;
+
+            NodeId createNodeId = null;
+            QualifiedName createQualifiedName = new QualifiedName(alarmName, NamespaceIndex);
+            LocalizedText createLocalizedText = null;
+
+
+            bool isBranch = IsBranch(branchId);
+            createNodeId = new NodeId(alarmNodeId, NamespaceIndex);
+            createLocalizedText = new LocalizedText(alarmName);
+
+            alarm.ReferenceTypeId = ReferenceTypeIds.HasComponent;
+            alarm.Create(
+                SystemContext,
+                createNodeId,
+                createQualifiedName,
+                createLocalizedText,
+                true);
+
+
+            if (!isBranch)
+            {
+                m_trigger.AddReference(ReferenceTypes.HasCondition, false, m_alarm.NodeId);
+                m_parent.AddChild(alarm);
+            }
+
+        }
+
+        private bool IsBranch(NodeId branchId)
+        {
+            bool isBranch = false;
+            if (branchId != null && !branchId.IsNullNodeId)
+            {
+                isBranch = true;
+            }
+            return isBranch;
+        }
+
+        public NodeId GetNewBranchId()
+        {
+            return new NodeId(++m_branchCounter, NamespaceIndex);
+        }
+
+        public virtual string GetBranchNodeIdString(NodeId branchId)
```

Comment:
This method is not called from anywhere else than line 99 which is commented; Shouldn't this method be removed  ?